DiabloII.Application/Controllers/NotificationsController.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using DiabloII.Application.Extensions;
+using DiabloII.Application.Requests.Notifications;
+using DiabloII.Application.Responses.Notifications;
+using DiabloII.Domain.Commands.Notifications;
+using DiabloII.Domain.Handlers;
+using DiabloII.Domain.Readers;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DiabloII.Application.Controllers
+{
+    [Route("api/v1/")]
+    public class NotificationsController : Controller
+    {
+        private readonly INotificationCommandHandler _handler;
+       
+        private readonly INotificationReader _reader;
+
+        private readonly IMapper _mapper;
+
+        public NotificationsController(INotificationCommandHandler handler, INotificationReader reader, IMapper mapper)
+        {
+            _handler = handler;
+            _reader = reader;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Get all the notifications
+        /// </summary>
+        [Route("notifications")]
+        [HttpGet]
+        [ProducesResponseType(typeof(IReadOnlyCollection<NotificationDto>), StatusCodes.Status200OK)]
+        public ActionResult<IReadOnlyCollection<NotificationDto>> GetAll()
+        {
+            var response = _reader
+                .GetAll()
+                .Select(_mapper.Map<NotificationDto>)
+                .ToList();
+
+            return Ok(response);
+        }
```

Comment:
My architecture looks to have some duplicate, I think we can extract those flow in some templated methods